core/src/main/java/hudson/model/TopLevelItemDescriptor.java

```diff
@@ -114,6 +127,64 @@ public String getDisplayName() {
     }
 
     /**
+     * A description of this kind of item type. This description can contain HTML code but it is recommend to use text plain
+     * in order to avoid how it should be represented.
+     *
+     * @return A string, an empty string by default.
+     *
+     * @since TODO
+     */
+    @Nonnull
+    public String getDescription() {
+        try {
+            WebApp webapp = WebApp.getCurrent();
+            MetaClass meta = webapp.getMetaClass(this);
+            Script s = meta.loadTearOff(JellyClassTearOff.class).findScript("newInstanceDetail");
+            if (s == null) {
+                return "";
+            }
+            DefaultScriptInvoker dsi = new DefaultScriptInvoker();
+            StringWriter sw = new StringWriter();
+            XMLOutput xml = dsi.createXMLOutput(sw, true);
+            dsi.invokeScript(Stapler.getCurrentRequest(), Stapler.getCurrentResponse(), s, this, xml);
+            return sw.toString();
+        } catch (Exception e) {
+            return "";
+        }
+    }
+
+    /**
+     * Used to categorize this kind of item type. @see {@link ItemCategory}
+     *
+     * @return A string with the category identifier, {@link ItemCategory.UncategorizedCategory#getId()} by default.
+     *
+     * @since TODO
+     */
+    @Nonnull
+    public String getCategoryId() {
+        return new ItemCategory.UncategorizedCategory().getId();
+    }
+
+    /**
+     * @since TODO
```

Comment:
@jglick I've added this javadoc:

```
/**
 * Represents a file path pattern to get the Item icon in different sizes.
 *
 * For example: plugin/plugin-shortname/images/:size/item.png, where {@code :size} represents the different
 * icon sizes used commonly in Jenkins project: 16x16, 24x24, 32x32 or 48x48
 *
 * @return A string or null if it is not defined.
 *
 * @since TODO
 */
```

This change was pushed.